test/unit/table_deleted_marker_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class TableDeletedMarkerTest < Test::Unit::TestCase
+  def described_class
+    CouchdbToSql::TableDeletedMarker
+  end
+
+  def setup
+    @database = create_database
+    @changes = mock
+    @changes.stubs(:database).returns(@database)
+    @changes.stubs(:schema).returns(CouchdbToSql::Schema.new(@database, :items))
+    @changes.stubs(:log_info)
+    @handler = CouchdbToSql::DocumentHandler.new(@changes)
+    @handler.document = { '_id' => '12345' }
+  end
+
+  def test_init
+    @row = described_class.new(@handler, 'items')
+    assert_equal @row.primary_key, :item_id
+  end
+
+  def test_init_override_primary_key
+    @row = described_class.new(@handler, 'items', primary_key: 'foo_item_id')
+    assert_equal @row.primary_key, :foo_item_id
+  end
+
+  def test_handler
+    @row = described_class.new(@handler, :items)
+    assert_equal @row.handler, @handler
+  end
+
+  def test_key_filter
+    @row = described_class.new(@handler, :items)
+    assert_equal @row.key_filter, item_id: '12345'
+  end
+
+  def test_execution_deletes_rows
+    @database[:items].insert(name: 'Test Item 1', item_id: '12345')
+    assert_equal @database[:items].count, 1, 'Did not create sample row correctly!'
+    @row = described_class.new(@handler, :items)
+    # @row.execute # TODO: Sqlite and PostGres have different upsert syntax
+    assert_equal 1, @database[:items].count
+    # assert_true @database[:items].first._deleted
+  end
+
+  protected
+
+  def create_database
+    database = Sequel.sqlite
+    database.create_table :items do
+      String :item_id
+      String :name
+      Time :created_at
+      index :item_id, unique: true
+    end
+    database.create_table :groups do
+      String :group_id
+      String :name
+      Time :created_at
+      index :group_id, unique: true
+    end
+    database
+  end
+end
```

Comment:
Thanks for _trying_ to add tests here! We'll have to complete it some other time if we are unable to find a good way to complete it now.

Reply:
Just for the record here why Mr @perlun said _trying_. The tests are running on Sqlite, while we run our real code on Postgres. The syntax for upserts (insert_conflict) is different for Sqlite and Postgres.

Reply:
Indeed. So the only way to properly test it right now would be to use a real Postgres database for testing. It's _doable_, but it mean we would have to add Rake tasks for setting it up, run it on Travis, etc etc.